consensus/src/liveness/proposal_generator.rs

```diff
@@ -195,12 +194,26 @@ impl PipelineBackpressureConfig {
                     }
                 })
                 .sorted()
-                // .sorted_by_key(|key| key.unwrap_or(u64::MAX))
```

Comment:
As we are trying to calculate the `max_txns_after_filtering` here, we are interested in the number of transactions that are actually committed.
So, should the formula be 
```
                  ((config.target_block_time_ms as f64 / execution_time_ms as f64
                                * summary.to_commit as f64)
                                .floor() as u64)
                                .max(1),
                        )
```
instead of
```
                ((config.target_block_time_ms as f64 / execution_time_ms as f64
                                * (summary.to_commit as f64
                                    / (summary.to_commit + summary.to_retry) as f64)
                                * summary.payload_len as f64)
                                .floor() as u64)
                                .max(1),
                        )
```

Reply:
we are only able to set max_txns_after_filtering/max_txns_to_execute, and we can have bunch of txns being discarded (in addition to to_commit and to_retry). 

so if 30% is to_commit, 20% is to_discard and 50% to_retry

then if block was 1000 transactions, to_commit is 300, and above formula sets it to 375, to allow for discards.

but everything here is an approximation. after execution pool, we should just set block gas limit, and there wouldn't be an issue like this.